program/rust/src/utils.rs

```diff
@@ -233,18 +251,15 @@ pub fn check_is_upgrade_authority_for_program(
 
 
     // 2. upgrade_authority_account is actually the authority inside programdata_account
-    if let UpgradeableLoaderState::ProgramData {
-        slot: _,
-        upgrade_authority_address: Some(upgrade_authority_key),
-    } = programdata_deserialized
-    {
-        pyth_assert(
-            upgrade_authority_key.eq(upgrade_authority_account.key),
-            OracleError::InvalidUpgradeAuthority.into(),
-        )?;
-    } else {
-        return Err(OracleError::InvalidUpgradeAuthority.into());
-    }
+    pyth_assert(
+        programdata_deserialized
+            .upgrade_authority
+            .eq(upgrade_authority_account.key)
+            && programdata_deserialized.account_type == 3
```

Comment:
what’s the account type check mean? (maybe leave a comment)

Reply:
ProgramdataAccount comments explain this.